### PR TITLE
Req posts

### LINKED ISSUE #1: Basic features of the app
This is a derivative from shift-board/shiftboard#5. Rather than creating raw HTML files, it will be more helpful to create a single page AJAX app using React.

These are the points from shift-board/shiftboard#5:

- The main board, with posts.
  - The title and description should be present alongside a title bg, above the posts.
  - An add button is present to route the user to the add page. (frontend routing now)
  - Each post should be able to expand if there is a message to display the message and the author.
  - If a post does not have a message, it should expand to show the author. The image should be bigger to account for this.
  - If a post only has a message, it can still expand. The author's name should be present, and the message should be made to look attractve.
  - When a post expands, a popup will appear over the posts and prominently display the selected image/message alongside the author of the post.
  - For mobile pages, forgo expansion but display each author's name above each post. The message can be collapsable, like a caption or description.
  - An "Add post" page for each board.
  - This should have an input box for the message, author, and a spot to upload an image.
  - A send button to send a POST request to the correct view/URL for handling.
- Additionally, the following must be kept in mind:
  - ~The proper variable names are used in the templates.~
  - The site is responsive and can resize. If the screen is large or small, the columns will resize to adjust.
  - Both pages have a functional mobile port.
  - The images should not all be loaded in at once. Only the first x amount of images should be loaded, and then the web page can request more images.
  
  
Task list:

- [x] Set up the React template along with corresponding dependencies
- [ ] Create the different pages + components
- [x] Set up the routing system
- [ ] Use the Fetch API and get data from the Django backend -> https://github.com/shift-board/shiftboard
- [ ] Create production build files and test it with the backend

### PR DESCRIPTION
**Pull request by shari09**
Able to fire the request post function as the user scrolls to the bottom. It also stops requesting if the amount of retrieved posts is less than the amount requested.

### CODE REVIEW

File: src/components/Navigation.tsx [hunk 1]
```diff
@@ -19,9 +20,79 @@ import {Link} from 'react-router-dom';
...
+
+  // The other navbar items excluding the home button.
+  //TODO: figure out the necessity of `useMemo` here
+  const navItems = useMemo(() => {
+    const addStyle: SxStyleProp = {
+      fontSize: 'large',
+    };
+
+    const links = {
+      '/add-post': <BsPlusCircleFill sx={addStyle}/>,
```
Comment (wang-joseph) [inline src/components/Navigation.tsx]: `add` has triumphed!!!

File: src/pages/Home.tsx [hunk 2]
```diff
@@ -21,10 +25,164 @@ import {jsx, SxStyleProp} from 'theme-ui';
  * @returns The home page. 
  */
 export const Home: React.FC = () => {
+  //the board
+  const board = useContext<IBoardContext>(BoardContext);
+  const [posts, setPosts] = useState<IPost[]>([]);
+  const [fullyLoaded, setFullyLoaded] = useState<boolean>(false);
+  const [index, setIndex] = useState<number>(0);
```
Comment (wang-joseph) [inline src/pages/Home.tsx]: Docs for these variables, maybe?

File: src/pages/Home.tsx [hunk 3]
```diff
@@ -21,10 +25,164 @@ import {jsx, SxStyleProp} from 'theme-ui';
...
+      },
+      {
+        name: 'shari',
+        message: `hi sift if u didn't know already i actually love u so much. 
+im truly blessed to have met u back then, and im also rlly happy that we got closer. ur responsible and cool
+and u do a lot of cool things. but most of all, idk why, i just smile wheenver i look at u. i wanna see u happy,
+i wanna make u happy. i wanna stay forever with you. when i think about u, i just think of a smiling u, and its soooooo
+ooooooooo cute ^_^ i want u to find me cute, i want u to like me. u make me better and make my life better. u stick
+with me and we work things out well. our humours match and we're always on the same frequency. i wanna stick with u,
+ur such an amazing person and an amazing boyfriend (ur mine >:)) that is just too rare. i love u and i'll always do so ;)`,
```
Comment (wang-joseph) [inline src/pages/Home.tsx]: i will never ever get tired of reading such an amazing message from such an amazing girlfriend

Comment (wang-joseph) [reply src/pages/Home.tsx]: 😊

File: src/pages/Home.tsx [hunk 4]
```diff
@@ -21,10 +25,164 @@ import {jsx, SxStyleProp} from 'theme-ui';
...
+  };
+
+
+  /**
+   * Loads 30 more posts from the server.
+   */
+  const loadMorePosts = () => {
+    const amount = 30; // To test infinite scrolling, change this to like 2 (getMorePost returns 3 posts right now)
+    const newPosts = getMorePosts(index, amount);
+    if (newPosts.length < amount) {
```
Comment (wang-joseph) [inline src/pages/Home.tsx:100]: Could this ever be less than or equals to?

**Review by wang-joseph — CHANGES_REQUESTED** [review]

File: src/pages/Home.tsx (hunk 3 above)
Comment (shari09) [reply src/pages/Home.tsx]: 😛 😛 😛 😛 😛

Comment (shari09) [reply src/pages/Home.tsx]: this will be purged soon tho 😉

File: src/pages/Home.tsx (hunk 4 above)
Comment (shari09) [reply src/pages/Home.tsx:100]: yes, if the database didn't return back 30 posts since it ran out

File: src/pages/Home.tsx (hunk 2 above)
Comment (shari09) [reply src/pages/Home.tsx]: sure

**Comment by shari09** [comment]
i resolved the merge conflict it with master 😄

File: src/pages/Home.tsx (hunk 3 above)
Comment (wang-joseph) [reply src/pages/Home.tsx]: never from my heart

File: src/pages/Home.tsx (hunk 4 above)
Comment (wang-joseph) [reply src/pages/Home.tsx:100]: Ah hold on I finally got what that meant sorry my bad am monkee brain

File: src/components/home/Post.tsx [hunk 5]
```diff
@@ -146,12 +146,12 @@ export const Post: React.FC<PostProps> = React.memo(({post, style}) => {
             setMessageHeight(element.getBoundingClientRect().height);
             if (element.scrollHeight > element.getBoundingClientRect().height) {
               //TODO: fix this bug of overflow detection
-              console.log(element.scrollHeight, element.getBoundingClientRect().height);
+              console.log('FIX THIS BUG', element.scrollHeight, element.getBoundingClientRect().height);
```
Comment (wang-joseph) [inline src/components/home/Post.tsx:149]: FIX THIS BUG

**Review by wang-joseph — APPROVED** [review]
LGTM

**Pull request merged**